06-11-2018/Italent2BoardCreation/src/main/java/com/italent2/service/BoardCreationAndMessagePosting.java

```diff
@@ -0,0 +1,141 @@
+import org.apache.http.client.HttpResponseException;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.BasicResponseHandler;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.italent2.model.Messages;
+
+public class BoardCreationAndMessagePosting 
+{
+	ResourceBundle rb=ResourceBundle.getBundle("Application");
+	public String sesiionKeyGeneration() throws HttpResponseException, IOException
+	{
+			CloseableHttpClient client = HttpClients.createDefault();
+			HttpPost postRequest=new HttpPost(rb.getString("posturl"));
+			postRequest.addHeader("Authorization","Basic ZGVtbzAxOlN1UHIxKm1aTGk=");
+			HttpResponse response=client.execute(postRequest);
+			int statusCode=response.getStatusLine().getStatusCode();
+			if(statusCode!=200)
+			{
+				throw new RuntimeException("failed with response code"+statusCode);
+			}
+			System.out.println(response);
+			System.out.println("generating a session key");
+			ObjectMapper objectMapper = new ObjectMapper();
+			JsonNode tree = objectMapper.readTree(new BasicResponseHandler().handleResponse(response));
+			JsonNode response1=tree.get("response");
+			JsonNode value=response1.get("value");
+			String key=value.get("$").asText();
+			System.out.println("Generated session key is: "+key);
+			return key;
+	}
+	
+	public void boardCreation(String key)
+	{
+		try
+		{
+			CloseableHttpClient client = HttpClients.createDefault();
+			HttpPost postRequest=new HttpPost(rb.getString("boardurl"));
+			List<NameValuePair> params = new ArrayList<NameValuePair>();
+			params.add(new BasicNameValuePair("restapi.session_key",key.toString()));
+			params.add(new BasicNameValuePair("restapi.response_format","json"));
+			postRequest.setEntity(new UrlEncodedFormEntity(params));
+			postRequest.addHeader("Authorization","Basic ZGVtbzAxOlN1UHIxKm1aTGk=");
+			HttpResponse response=client.execute(postRequest);
+			int statusCode=response.getStatusLine().getStatusCode();
+			if(statusCode!=200)
+			{
+				throw new RuntimeException("failed with response code"+statusCode);
+			}
+			System.out.println("successfully created the board");
+		}catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+	
+	public HttpResponse setMessage(String key) throws ClientProtocolException, IOException
+	{
+		CloseableHttpClient client = HttpClients.createDefault();
+		HttpPost postRequest=new HttpPost(rb.getString("messageurl"));
+		List<NameValuePair> params = new ArrayList<NameValuePair>();
+		params.add(new BasicNameValuePair("restapi.session_key",key.toString()));
+		params.add(new BasicNameValuePair("restapi.response_format","json"));
+		postRequest.setEntity(new UrlEncodedFormEntity(params));
```

Comment:
All credentials should be fetch from properties file .remove this "Basic ZGVtbzAxOlN1UHIxKm1aTGk="